### DISCUSSION TITLE
Info e consigli per passaggio da raspberry a mini pc

### DISCUSSION THREAD

**Discussion by blakdog1978** [indomus/forum#2621] [HUB: Home Assistant]
Buongiorno,
ho acquisato questo mini pc:
Blackview Mini PC Intel Alder Lake Ν95 12th (fino a 3,4 GHz), 16GB DDR4 512GB
e vorrei un consiglio sulle possibilità di installazione di hassios su di esso,
ho dato un'occhiata alle guide ufficiali e opterei di installare Home Assistant OS,
ma mi chiedevo se c'è un modo per avere Assistant os mantenendo come sistema opeativo windows sul mini pc.
Se fosse possibile mi chiedo di conseguenza se le prestazioni sarebbero minori visto che vorrei far girare Frigate, Double Take e Deepstack per rilevamento dei volti.
Grazie a tutti.

**Comment by snakuzzo** [comment]
Ciao.
Puoi creare una VM con Virtualbox o simili senza alcun problema a livello di risorse.
Leggi [QUI](https://www.home-assistant.io/installation/alternative#download-the-appropriate-image)

**Reply by blakdog1978** [reply]
ok grazie ,
proverò.

**Reply by blakdog1978** [reply]
ma secondo te posso usare un backup di assio del raspberry e usarlo sul mini pc sempre con assio installandolo come sistema operativo non in VM ?

**Comment by KC-inDomus** [comment]
> Buongiorno, ho acquisato questo mini pc: Blackview Mini PC Intel Alder Lake Ν95 12th (fino a 3,4 GHz), 16GB DDR4 512GB e vorrei un consiglio sulle possibilità di installazione di hassios su di esso, ho dato un'occhiata alle guide ufficiali e opterei di installare Home Assistant OS, ma mi chiedevo se c'è un modo per avere Assistant os mantenendo come sistema opeativo windows sul mini pc. Se fosse possibile mi chiedo di conseguenza se le prestazioni sarebbero minori visto che vorrei far girare Frigate, Double Take e Deepstack per rilevamento dei volti. Grazie a tutti.

secondo me lasciare in piedi windows non ha alcun senso. se vuoi far girare diverse cose, evita hassos su windows e hassos in generale: tira su uno stack linux debian 12 con docker e divertiti.

**Reply by blakdog1978** [reply]
Ci potrei pensare ma non so se sarei in grado

**Reply by KC-inDomus** [reply]
prova: al massimo torni indietro

**Comment by Newreader** [comment]
Mi inserisco anch'io per chiedere il vostro aiuto e consiglio. A dicembre 2025 i sistemi a 32 bit non saranno più supportati per cui ho deciso di passare ad mini pc.
Riepilogo per sommi capi la mia architettura. Ho un Raspberry 3B+ con SO Raspbian al quale è collegato attraverso una porta USB un SSD con Home Assistant. All'accensione HA parte in automatico. Sul Raspberry ho alcune Installazioni con docker: home-assistant, Zigbee2mqtt per gestire i dispositivi zigbee, DuckDns+Let’s Encrypt+ Caddy come reverse proxy per gestire gli accessi in remoto Nut-upsd per gestire l’UPS, Broadlink Manager per gestire i dispositivi broadlink, Eclipse-masquitto per gestire l'interfacciamento mqtt wifi. Gestisco il sistema attraverso integrazioni messe a disposizione da HA e pacchetti yaml scritti da me a partire da configuration.yaml, automations.yaml, scripts.yaml e altro. Ho installato il custom component HACS. Mi sono creato una skill per Alexa con la finalità di far “vedere” ad Alexa dispositivi visti da Home Assistant che diversamente non potrebbe vedere e conseguentemente comandarli con Alexa (anche vocalmente). Ho dispositivi shelly e sonoff con tasmota, due telecamere che gestisco solo ed esclusivamente con Home Assistant. Ho montato il sistema Tado (termostato+teste intelligenti con sensori di temperatura esterni). Ho un sisteam per controllare che le apparecchiature/dispositivi non assorbano più dei watt consentiti per un cero tempo nel qual caso il sistema autonomamente (attraversoo file .yaml e .py)  stacca il dispositivo che consuma di più dandone contezza su telegram dove ho creato un bot personalizzato, su HA Comapanion e su Alexa.   Il sistema è governato da un router wifi TP-Link C1200 che è collegato ad modem Fritz!Box 5590 (ho operato questa scelta perché inizialmente il TP-Link funzionava anche come modem poi sostituito dal Fritz!box per cui non volendo cambiare tutti i riferimenti IP dei dispositivi collegati ho lasciato il TP-Link come router wifi.
Ho acquistato un mini pc sul quale ho installato debian 12 oltre che il container Cloudflare che deve sostituire DuckDns+Let’s Encrypt+ Caddy. Vi chiedo qual è secondo voi il metodo meno oneroso per passare dal Raspberry al mini pc mantenedo il più possibile l’attuale configurazione (configuration.yaml, automations.yaml, scripts.yaml, ui-lovelace.yaml e gli altri file  .yaml) (entità, device, integrazioni e così via)
Grazie anticipatamente

**Reply by KC-inDomus** [reply]
Innanzitutto sappi che domani esce Debian 13, quindi partirei da un'installazione pulita.
Dopodiché: fatti un backup delle cartelle alle quali puntano i tuoi container, riproducile sul sistema target e, tramite docker compose, ripristina tutto. Non dovresti avere grossi problemi a ripristinare il tuo stack, è un lavoro di mezz'ora (al netto di tirare su la macchina target con Debian+Docker).

**Reply by Newreader** [reply]
Grazie

**Reply by Newreader** [reply]
Ho provato a seguire il tuo consiglio e ho iniziato con il container zigbee2mqtt dopo aver ovviamente creato il container mosquitto.
Il mio zigbee2mqtt è stato creato originariamente sul Rapberry sotto la directory /opt per cui ho copiato tutto ciò che sta dentro /opt/zigbee2mqtt dal raspberry al mini pc. Dopo ho posizionato i due dispositivi su due reti diverse (raspberry TP-Link: 192.168.0.xxx) (mini pc Frit!Box: 192.168.178.yyy) perché notoriamente due coordinatori non possono stare sulla stessa rete.
Ho adattato ovviamente il configuration.yaml perché il server mqtt adesso per questi test si trova sulla rete Fritz!Box e ho utilizzaato una pennetta ZBDongle Sonoff ver. 2 E mentre sul Raspbery ho la ver1. P) quindi ho modificato anche il serial: sul configuration.yaml.
Ho fatto le verifiche per controllare se la pennetta di cui ho anche aggiornato il software fosse rilevata dal sistema, in ambedue i casi è vista come /dev/ttyUSB0 ho controllato i permessi sul device:
Il device /dev/ttyUSB0 ha permessi corretti
Il proprietario è root
Il gruppo è dialout
Il mio utente fa parte del gruppo dialout
A questo punto con docker compose ho ricreato il container ma purtroppo mi da errore:
z2m: Error: USB adapter discovery error (No valid USB adapter found). Specify valid 'adapter' and 'port' in your configuration
Sono andato sul sito i zigbee2mqtt e ho trovato tra le faq che in questi casi occorre modificare il devices in docker-compose.yaml, cosa che puntualmente ho fatto inserendo:
- "/dev/serial/by-id/usb-Itead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_V2_60e6dc49f673ef1188fec274d9b539e6-if00-port0:/dev/ttyUSB0"
ma l'errore permane.
Ho fatto diversi test di cui tra gli altri:
se creo tutto ex-novo non ho nessun problema
ho testato i file database.db e state.json presi dalla installazione problematica e inserendoli nella installazione ex-novo e non ho avuto problemi.
Potreste indicarmi soluzioni a questo problema magari da parte di qualcuno che ha avuto i miei stessi problemi e li ha risolti?
